File: sdks/dotnet/sulfone-helium/Domain/Plugin/Service.cs
using sulfone_helium.Domain.Core;

namespace sulfone_helium.Domain.Plugin;

public class PluginService
{
    public required ICyanPlugin Plugin { private get; init; }

    public async Task<PluginOutput> Plug(PluginInput input)
    {
        var (dir, config) = input;
        return await Plugin.Plugin(new CyanPluginInput { Directory = dir, Config = config });
    }
}

File: sdks/python/cyan_sdk/domain/plugin/service.py
from cyan_sdk.domain.core.cyan_script import ICyanPlugin
from cyan_sdk.domain.core.cyan_script_model import CyanPluginInput
from cyan_sdk.domain.plugin.input import PluginInput
from cyan_sdk.domain.plugin.output import PluginOutput


class PluginService:
    def __init__(self, plugin: ICyanPlugin):
        self._plugin = plugin  # Leading underscore indicates a "private" attribute

    async def plug(self, i: PluginInput) -> PluginOutput:
        directory = i.directory
        config = i.config
        return await self._plugin.plugin(CyanPluginInput(directory=directory, config=config))
